code-challenges.test.js

```diff
@@ -21,8 +23,51 @@ const secretCodeWord2 = "Gobbledygook"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+describe ('coder', () => {
+    it ('takes in a string and returns it coded', () => {
+        expect (coder(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect (coder(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect (coder(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
+// FAIL  ./code-challenges.test.js
+// coder
+//   ✕ takes in a string and returns it coded
+
+// ● coder › takes in a string and returns it coded
+
+//   ReferenceError: coder is not defined
+
 // b) Create the function that makes the test pass.
 
+// PSEUDOCODE
+// create fxn called coder that takes in a string and returns it coded
+// input : string
+// output : string with a as 4, e as 3, i as 1, and o as 0
+// loop through string using for loop
+// if i === e or E, i = 3
+
+const coder = (str) => {
+    string = str.split('')
+    for (i=0; i < string.length; i++) {
+        if (string[i] === 'e' || string[i] === 'E') {
+            string.splice(i,1,'3')
+        } else if (string[i] === 'a' || string[i] === 'A') {
+            string.splice(i,1,'4')
+        } else if (string[i] === 'i' || string[i] === 'I') {
+            string.splice(i,1,'1')
+        } else if (string[i] === 'o' || string[i] === 'O') {
+            string.splice(i,1,'0')
+        }
+    }
+    return string.join('')
+}
+
+// PASS  ./code-challenges.test.js
+// coder
+//   ✓ takes in a string and returns it coded (1 ms)
```

Comment:
Great structure, pseudocode, and logic. Note: When you see that you are repeating the same logic, check for a high order function that can lift that load. 
A recommended google search `javascript how to replace characters in a string`